### PR TITLE
Drop smithy files from smithy4s-client

### PR DESCRIPTION
**Pull request by kubukoz**
## Changes Introduced

- remove smithy4s codegen plugin from modules that don't need it
- make `smithy4s-client` a non-smithy-library. This will result in the jar not containing a Smithy manifest and copies of the codegen sources.

Jar contents before:

image

Jar contents after:

image

## Checklist (check all that apply)

- [x] This change maintains backwards compatibility
- [x] I have introduced tests for all new features and changes
- [x] I have added documentation covering all new features and changes
- [x] This pull-request is ready for review

### CODE REVIEW

**Review by etspaceman — APPROVED** [review]

File: build.sbt [hunk 1]
```diff
@@ -124,7 +124,6 @@ lazy val kcl = projectMatrix
   .dependsOn(shared, `kinesis-client`)
 
 lazy val `kcl-http4s` = projectMatrix
-  .enablePlugins(Smithy4sCodegenPlugin)
```
Comment (etspaceman) [inline build.sbt:127]: This one needs to stay because we are referencing the generated classes from it.

**Review by etspaceman — CHANGES_REQUESTED** [review]
Re-add to kcl-http4s

File: build.sbt (hunk 1 above)
Comment (kubukoz) [reply build.sbt:127]: The plugin shouldn't be needed for accessing generated classes, I'll have to dive deeper to see why they're not available there :|

Comment (kubukoz) [reply build.sbt:127]: AH. this module has smithy files itself, I see.

**Comment by kubukoz** [comment]
think it's all good now :)

**Review by etspaceman — APPROVED** [review]

**Pull request merged**